### PR TITLE
[MM-9321] Add a "+" button to make it easy to add Emoji Reactions

### PR DESCRIPTION
**Pull request by cpanato**
#### Summary
Add a "+" button to make it easy to add Emoji Reactions

#### Ticket Link
Resolves: #1293
Jira: https://mattermost.atlassian.net/browse/MM-9321

#### Checklist
- [x] Has UI changes

#### Device Information
This PR was tested on: IOS, iphone 8 plus

#### Screenshots
m3xxsesboq

### CODE REVIEW

File: app/components/post_body/post_body.js [hunk 1]
```diff
@@ -296,7 +296,7 @@ class PostBody extends PureComponent {
                             isReplyPost={isReplyPost}
                         />
                         {this.renderFileAttachments()}
-                        {hasReactions && <Reactions postId={postId}/>}
+                        {hasReactions && <Reactions postId={postId} onAddReaction={this.props.onAddReaction} />}
```
Comment (enahum) [inline app/components/post_body/post_body.js]: you have some eslint errors here, to fix do:

```js
{hasReactions &&
<Reactions
    postId={postId}
    onAddReaction={this.props.onAddReaction}
/>
}
```

File: app/components/reactions/reactions.js [hunk 2]
```diff
@@ -5,9 +5,13 @@ import React, {PureComponent} from 'react';
 import PropTypes from 'prop-types';
 import {
     StyleSheet,
-    View
+    View,
+    TouchableOpacity,
```
Comment (enahum) [inline app/components/reactions/reactions.js]: can we sort these?

File: app/components/reactions/reactions.js [hunk 3]
```diff
@@ -20,7 +24,8 @@ export default class Reactions extends PureComponent {
         highlightedReactions: PropTypes.array.isRequired,
         postId: PropTypes.string.isRequired,
         reactions: PropTypes.object.isRequired,
-        theme: PropTypes.object.isRequired
+        theme: PropTypes.object.isRequired,
+        onAddReaction: PropTypes.func.isRequired
```
Comment (enahum) [inline app/components/reactions/reactions.js]: also this one, can you move it so it is sorted?

File: app/components/reactions/reactions.js [hunk 4]
```diff
@@ -55,9 +60,24 @@ export default class Reactions extends PureComponent {
     }
 
     render() {
+        const {reactions}= this.props;
+        const styles = getStyleSheet(this.props.theme);
+        let addMoreReactions = null;
+        if (reactions.size != 0) {
```
Comment (enahum) [inline app/components/reactions/reactions.js]: perhaps is best if you do the following

```js
if (!reactions.size) {
    return null;
}

const addMoreReactions = (
    <TouchableOpacity
         onPress={this.props.onAddReaction}
        style={[styles.reaction]}
    >
       <Text style={styles.more}>{'+'}</Text>
   </TouchableOpacity>
);
```

and then leave the return as it is.

so if no reactions it just return and if you have at least one reaction then it shows you the `+` option

File: app/components/post_body/post_body.js (hunk 1 above)
Comment (cpanato) [reply app/components/post_body/post_body.js]: done

File: app/components/reactions/reactions.js (hunk 2 above)
Comment (cpanato) [reply app/components/reactions/reactions.js]: done

File: app/components/reactions/reactions.js (hunk 3 above)
Comment (cpanato) [reply app/components/reactions/reactions.js]: done

File: app/components/reactions/reactions.js (hunk 4 above)
Comment (cpanato) [reply app/components/reactions/reactions.js]: done

thanks so much for your review and support @enahum

**Comment by enahum** [comment]
Awesome @cpanato thanks for the PR

**Pull request merged**